### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by httpsnet**
Again, we are open for discussion and fix commits.
There are many corrections and changes this time, but the specification changes have been suppressed.
I also think there will be a lot of conflicts in PR. sorry.

Fixed an issue where Achievements 63, 75, 76, 77 and 247 could not be earned.
Fixed the #sacrificeAnts box sometimes overlapping Ant Tier upgrades and not pressing.
Fixed Ascend History and Singularity History layouts.
Updated CSS.
Added $STAGES$. This shows the current stage including Singularity.
Fixed an issue with Last enabled key not working.
Ascension and Ascension Challenges can no longer be completed without Achievements 141. This could work with things like Auto Challenge Sweep and hotkey and HTML editing exploits.
Added Singularity Count to Miscellaneous Stats.
Added Auto Ascend, Open 10% All Cubes, Open 50% All Cubes and Open 100% All Cubes to Hotkeys.
The Hotkeys list is more visually appealing.
Hotkeys list now shows Description on hover. The text is inadequate.
Numeric hotkeys are also disabled when the dialog is visible.
If the same key is set for hotkey and number hotkey, they will not work at the same time.
If the key setting prompt is open and the textbox is not focused, typing will write the key to the textbox.
Changed EventListener registration for eventHotkeys() to occur at the end of document.onload.
Enabled revealStuff() on autoupgrades if you haven't unlocked Ascension.
Singularity upgrades free levels in daily codes are no longer numbered.
Fixed Talisman Shards Auto Buy being able to function without being unlocked.
Updated r3x21-r3x25 effects.
Updated internal code for Toggle amount to buy. No feature additions at this time, but will be in the near future.
Saves the current game version that started the save data to the save data.
The Automations toggle allows you to control which upgrades you get on reset.
Shared shop potions and singularityQuarks elements.
Added Descriptions to singularityQuarks.
Cx1 is now automatically obtained upon ascension from Singularity 11.
Auto Challenge Sweep will be enabled from the start with Singularity 40.
Auto Ascend Mode: Real time reduced to Singularity 5.
Automation Shop is automatically reduced to Singularity 20 and added an Automation Shop toggle.
Autobuy Talismans Resources have been reduced to Singularity 25 and no longer require Cx1.
Added Auto Runes to Hotkeys.
Fixed an issue where Auto Generator Upgrade may not work.
Added lock function to hotkey. Hotkeys are disabled when functionality is restricted.
Replaced some loops from constants to proper array lengths. I think that this makes it less likely that problems will occur even if there is an upgrade addition.
Adding variables for loading many toggles and using them for initial settings of toggles helps reuse code and maintenance such as adding functionality.
Challenge Sweep no longer works unless Achievements 134 is achieved.
Auto Cube Manipulation no longer works unless Achievements 141 is achieved.
Layout Shift is no longer caused by scrollbars in WebKit.
Added class "buildingPurchaseBtnAvailable" if Crystal Upgrades are available for purchase.
Added Auto Cube Upgrades. It will be unlocked in Singularity 45.
Added a class "runeButtonAvailable" if it is possible to increase the Runes level by 1 or more.
Auto Runes will use Offerings if it is possible to raise the level of Runes by 1 or more.
Fixed updateChallengeDisplay() sometimes not working on reset.
Automation Shop is automatically reverted to Singularity 25.
You can now enable Rune Sacrifices even if you have w2x10.
Made the scrollbar track transparent.
Firefox now applies scrollbar colors.
I made the background color of the scrollbar transparent.
I used format() with less expensive code. format() now correctly displays extremely large numbers.
Adjusted CSS. Width must not exceed 1250px for mobile considerations.
Fixed an issue where preloadDeleteGame() did not work due to stacking order of confirmation screens.
Added Toggle Notation. This controls Notation.
Added toggle Top Stats. This controls the display of #ascensionStats.
Added Toggle Mini Header. This controls the size of #subHeader.
Added toggle Auto Export Quark. This automatically controls Export Quark.
Changed the display method of #ascensionStats.
The sentence "These do not work with hotkeys. Too bad!" has been removed as it is now meaningless.
I remade the research BG. This should fix the class switching problem.
#singularitybtn now shows Golden Quarks icon and number on hover.

### CODE REVIEW

**Comment by httpsnet** [comment]
I found a problem in research. Please wait until I fix it

File: src/Variables.ts [hunk 1]
```diff
@@ -242,6 +242,7 @@ export const Globals: GlobalVariables = {
     antEightProduce: new Decimal('1'),
 
     antCostGrowth: [1e41, 3, 10, 1e2, 1e4, 1e8, 1e16, 1e32],
+    antProducerCostVals: ['null', '1e700', '3', '100', '10000', '1e12', '1e36', '1e100', '1e300'],
 
     antUpgradeBaseCost: [100, 100, 1000, 1000, 1e5, 1e6, 1e8, 1e11, 1e15, 1e20, 1e40, 1e100],
     antUpgradeCostIncreases: [10, 10, 10, 10, 100, 100, 100, 100, 1000, 1000, 1000, 1e100],
```
Comment (mustache0110) [inline src/Variables.ts:248]: Mixing array index starting at 1 and 0 seems like asking for trouble

Comment (httpsnet) [reply src/Variables.ts:248]: understand. Revert changes to prevent confusion

**Comment by httpsnet** [comment]
Maybe better. I always thought it was the best patch, but I kept fooling myself every time I found a problem. I will continue to hate myself.
Marge is always welcome, but please review for safety.

Consume 10% of what you have when you shift and click to consume a potion.
Shift in purchasing potions and click to buy as much as possible.
Fixed an issue where some research backgrounds were not updating.
Fixed an issue where Auto Cube Upgrades would appear even at Singularity 45 and below.
Added Auto Platonic Upgrades. This works for Singularity 50 and above. Auto Platonic Upgrades ignore Offerings and Obtainium.
Fixed Platonic ALPHA...? not updating Platonic Upgrades backgrounds.
Fixed an issue where Hepteract Forge costs are multiplied not working. I also fix player friendly bugs for Plat's design.
Added 1 million to the number of purchases at one time. This will now appear in Singularity.
Singularity Upgrades and Octeract Upgrades can now be purchased at a time. This is visible at Singularity 50 and above.
Removed Shift and Click for Singularity Upgrades and Octeract Upgrades.
Removed the judgment that Auto Challenge Sweep does not work when Challenge 15 and Ascension Timer is less than 2. Probably not.

**Comment by httpsnet** [comment]
Maybe better. I always thought it was the best patch, but I kept fooling myself every time I found a problem. I will continue to hate myself.
Marge is always welcome, but please review for safety.

Consume 10% of what you have when you shift and click to consume a potion.
Shift in purchasing potions and click to buy as much as possible.
Fixed an issue where some research backgrounds were not updating.
Fixed an issue where Auto Cube Upgrades would appear even at Singularity 45 and below.
Added Auto Platonic Upgrades. This works for Singularity 50 and above. Auto Platonic Upgrades ignore Offerings and Obtainium.
Fixed Platonic ALPHA...? not updating Platonic Upgrades backgrounds.
Fixed an issue where Hepteract Forge costs are multiplied not working. I also fix player friendly bugs for Plat's design.
Added 1 million to the number of purchases at one time. This will now appear in Singularity.
Singularity Upgrades and Octeract Upgrades can now be purchased at a time. This is visible at Singularity 50 and above.
Removed Shift and Click for Singularity Upgrades and Octeract Upgrades.
Removed the judgment that Auto Challenge Sweep does not work when Challenge 15 and Ascension Timer is less than 2. Probably not.

**Comment by httpsnet** [comment]
For personal reasons this is closed.
If you're interested in this, you can do a partial PR.

**Pull request closed without merging**